mcs/class/Mono.Posix/Test/Mono.Unix.Native/SocketTest.cs

```diff
@@ -0,0 +1,215 @@
+//
+// socket-related test cases
+//
+// Authors:
+//  Steffen Kiess (s-kiess@web.de)
+//
+// Copyright (C) 2015 Steffen Kiess
+//
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+using Mono.Unix;
+using Mono.Unix.Native;
+
+using NUnit.Framework;
+
+namespace MonoTests.Mono.Unix.Native
+{
+	[TestFixture, Category ("NotDotNet")]
+	public class SocketTest {
+
+		string TempFolder;
+
+		[SetUp]
+		public void SetUp ()
+		{
+			TempFolder = Path.Combine (Path.GetTempPath (), this.GetType ().FullName);
+
+			if (Directory.Exists (TempFolder))
+				Directory.Delete (TempFolder, true);
+
+			Directory.CreateDirectory (TempFolder);
+		}
+
+		[TearDown]
+		public void TearDown()
+		{
+			if (Directory.Exists (TempFolder))
+				Directory.Delete (TempFolder, true);
+		}
+
+		// Set a timeout on all sockets to make sure that if a test fails it
+		// won't cause the program to hang
+		void SetTimeout (int socket)
+		{
+			Timeval timeout = new Timeval { tv_sec = 0, tv_usec = 500000 };
```

Comment:
Coding style: when using C# initialize syntax, please place each member on a separate line:

    var timeout = new Timeval {
        tv_sec  = 0,
        tv_usec = 500000,
    };

Also note the trailing `,`.